src/script/media/MediaDevicesHandler.ts

```diff
@@ -221,8 +273,18 @@ export class MediaDevicesHandler {
     const setDevices = (type: DeviceTypes, devices: MediaDeviceInfo[]): void => {
       this.availableDevices[type](devices);
       const currentId = this.currentDeviceId[type];
-      if (!devices.some(d => d.deviceId === currentId())) {
-        currentId(devices[0]?.deviceId);
+
+      const favorites = loadValue<string[]>(`${type}-fave`) ?? [];
+      const isFavorite = favorites.map(favorite => devices.some(d => d.groupId === favorite));
+
+      if (isFavorite.some(favorite => favorite)) {
+        const faveIndex = isFavorite.findIndex(favorite => favorite);
+        currentId(
+          devices.find(d => d.deviceId === favorites[faveIndex] || d.groupId === favorites[faveIndex])?.groupId ??
+            devices[0]?.groupId,
```

Comment:
devices[0]?.groupId has multiple occurrence. Using destructuring we can make it cleaner and more readable.
 
```
const [firstDevice = {}] = devices; 
const { groupId: firstDeviceGroupId } = firstDevice;
```

Reply:
i changed this slightly differently. this var is only used twice so i dont think this type of destructuring makes it more readable (to me)